### PR TITLE
Remove `feeCurrency`, `gatewayFee`, and `gatewayFeeRecipient` properties from Celo transaction receipts

### PR DESCRIPTION
**Pull request by 0xarthurxyz**
## Description

The `feeCurrency`, `gatewayFee`, and `gatewayFeeRecipient` fields are never included in a transaction receipt on Celo. 

See [here](https://github.com/celo-org/celo-blockchain/blob/master/internal/ethapi/api.go#L2067) where the transaction receipt is defined in the Celo blockchain client. 

Therefore, the 3 fields can be removed from transaction receipts across the `viem` library:

```diff
transactionReceipt: /*#__PURE__*/ defineTransactionReceipt({
    format(
      args: CeloRpcTransactionReceiptOverrides,
    ): CeloTransactionReceiptOverrides {
      return {
-        feeCurrency: args.feeCurrency,
-        gatewayFee: args.gatewayFee ? hexToBigInt(args.gatewayFee) : null,
-        gatewayFeeRecipient: args.gatewayFeeRecipient,
      }
    },
  }),
```

## GitHub Issue

- https://github.com/celo-org/viem/issues/5

---

### Focus of this PR:
This PR focuses on removing the `feeCurrency` property from the transaction receipt.

### Detailed summary:
- Removed the `feeCurrency` property from the transaction receipt in the `CeloRpcTransactionReceipt` and `CeloTransactionReceipt` types.
- Updated the `formattersCelo` object in the `celo/formatters.ts` file to remove the `transactionReceipt` formatter.
- Removed the `defineTransactionReceipt` import from the `celo/formatters.ts` file.
- Updated the `CeloRpcTransaction` and `CeloTransaction` types in the `celo/types.ts` file to remove the `CeloRpcTransactionReceiptOverrides` and `CeloTransactionReceiptOverrides` types.
- Updated the `getTransactionReceipt` import in the `public/getTransactionReceipt.test.ts` file.
- Removed the `getTransactionReceipt` test in the `public/getTransactionReceipt.test.ts` file.
- Updated the `chain` parameter in the `createPublicClient` function call in the `public/getTransactionReceipt.test.ts` file.
- Updated the `chain` parameter in the `createPublicClient` function call in the `chain w/ custom block type` test in the `public/getTransactionReceipt.test.ts` file.
- Updated the `chain` parameter in the `createPublicClient` function call in the `

> The following files were skipped due to too many changes: `src/actions/public/getTransactionReceipt.test.ts`, `src/chains/celo/formatters.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jxom** [comment]
Is this PR ready to review?

**Comment by 0xarthurxyz** [comment]
While you're here, I was curious if this [test](https://github.com/wagmi-dev/viem/actions/runs/6890201996/job/18742744412?pr=1500) is fixed on `main` (or is expected). 
It should be failing due to changes I made.

```diff
 FAIL  src/actions/public/getTransaction.test.ts > gets transaction (legacy)
Error: Snapshot `gets transaction (legacy) 4` mismatched

- Expected
+ Received

  {
    "blockHash": "0x89644bbd5c8d682a2e9611170e6c1f02573d866d286f006cbf517eec7254ec2d",
    "blockNumber": 15131999n,
-   "chainId": undefined,
+   "chainId": 1,
    "from": "0x47a6b2f389cf4bb6e4b69411c87ae82371daf87e",
    "gas": 200000n,
    "gasPrice": 57000000000n,
    "hash": "0x31a326e4190db96a0c12ef2f2aee6d4566635deb78a3c3497af208b8b7039f22",
    "input": "0xa9059cbb00000000000000000000000001cc0c6c6c57707f89ab0c9a0c22139c501ffba50000000000000000000000000000000000000000000000064cf7d4e7b36a8000",
    "nonce": 3,
    "r": "0x36b5c1f8f70e845a35784b4e18807af88c36fd4958bc71e7dab3b99293531f73",
    "s": "0x5e1a25ed16877580921003749f54d0a14981cd989c293c202a382f43d87b2c47",
    "to": "0x3d382228c54736d831fac2748f4734d9177c7332",
    "transactionIndex": 0,
    "type": "legacy",
    "typeHex": "0x0",
    "v": 37n,
    "value": 0n,
  }
```

**Comment by 0xarthurxyz** [comment]
> While you're here, I was curious if this [test](https://github.com/wagmi-dev/viem/actions/runs/6890201996/job/18742744412?pr=1500) is fixed on `main` (or is expected to fail). It shouldn't be failing due to changes I made.
> 
> ```diff
>  FAIL  src/actions/public/getTransaction.test.ts > gets transaction (legacy)
> Error: Snapshot `gets transaction (legacy) 4` mismatched
> 
> - Expected
> + Received
> 
>   {
>     "blockHash": "0x89644bbd5c8d682a2e9611170e6c1f02573d866d286f006cbf517eec7254ec2d",
>     "blockNumber": 15131999n,
> -   "chainId": undefined,
> +   "chainId": 1,
>     "from": "0x47a6b2f389cf4bb6e4b69411c87ae82371daf87e",
>     "gas": 200000n,
>     "gasPrice": 57000000000n,
>     "hash": "0x31a326e4190db96a0c12ef2f2aee6d4566635deb78a3c3497af208b8b7039f22",
>     "input": "0xa9059cbb00000000000000000000000001cc0c6c6c57707f89ab0c9a0c22139c501ffba50000000000000000000000000000000000000000000000064cf7d4e7b36a8000",
>     "nonce": 3,
>     "r": "0x36b5c1f8f70e845a35784b4e18807af88c36fd4958bc71e7dab3b99293531f73",
>     "s": "0x5e1a25ed16877580921003749f54d0a14981cd989c293c202a382f43d87b2c47",
>     "to": "0x3d382228c54736d831fac2748f4734d9177c7332",
>     "transactionIndex": 0,
>     "type": "legacy",
>     "typeHex": "0x0",
>     "v": 37n,
>     "value": 0n,
>   }
> ```

This is still failing on CI, but doesn't look like it's related to my changes.

**Comment by 0xarthurxyz** [comment]
> > Is this PR ready to review?
> 
> Yes, the main feature work is complete, but I wanted to address the [drop in code coverage](https://github.com/wagmi-dev/viem/pull/1500/checks?check_run_id=18742806991) before marking the PR as ready for review.
> 
> > 89.41% (-10.42%) compared to [1c0fe25](https://github.com/wevm/viem/commit/1c0fe25a7a83211e9592ae24a75efcba6718671b)

Code coverage seems to be dropping by about 10 percentage points in most PRs made by the public, so I assume this is expected and somehow related to CI. For example: 

- https://github.com/wevm/viem/pull/1592
- https://github.com/wevm/viem/pull/1499

**Comment by 0xarthurxyz** [comment]
@jxom this PR is ready for review

File: src/actions/public/getTransactionReceipt.test.ts [hunk 1]
```diff
@@ -50,38 +49,224 @@ test('gets transaction receipt', async () => {
...
-      "gatewayFee": null,
-      "gatewayFeeRecipient": undefined,
-      "logs": [],
+      "cumulativeGasUsed": 0n,
+      "effectiveGasPrice": 250000000n,
+      "from": "0xe665ced18b0998ede7236da308e311e9261dc984",
+      "gasUsed": 445896n,
+      "l1BatchNumber": 273767n,
+      "l1BatchTxIndex": 177n,
+      "l2ToL1Logs": [],
```
Comment (jxom) [inline src/actions/public/getTransactionReceipt.test.ts:71]: Should we instead include these properties on the transaction receipt formatter?

Comment (0xarthurxyz) [reply src/actions/public/getTransactionReceipt.test.ts:71]: I converted this from a Celo-specific test to a zkSync-specific test to preserve as much code coverage as possible. 

```diff
test('chain w/ custom block type', async () => {
  const client = createPublicClient({
-    chain: celo,
+    chain: zkSync,
    transport: http(),
  })
```

[Source](https://github.com/wevm/viem/pull/1500/files/05ceb2a9a9285588cc78a92cf775a417ba7d4227#diff-b03676226cedd362eeb6c685124b74a3d72a931d47004dba2128a463e810f2d0R52)

The reason is that Celo transaction receipts are no different than Ethereum transaction receipts. The previous transaction receipt formatter incorrectly added 3 fields (`feeCurrency`, `gatewayFee`, and `gatewayFeeRecipient`) that are not in the JSON-RPC response and thus ended up being `undefined` by default.

I considered removing the custom transaction receipt test, but instead chose to replace it with zkSync, which has a custom transaction receipt, to maintain code coverage.

Comment (jxom) [reply src/actions/public/getTransactionReceipt.test.ts:71]: Sorry, I got slightly confused!

**Pull request merged**